src/prefect/core/flow.py

```diff
@@ -469,9 +469,12 @@ def add_task(self, task: Task) -> Task:
             self.tasks.add(task)
             self._cache.clear()
 
-            case = prefect.context.get("case", None)
-            if case is not None:
-                case.add_task(task, self)
+            # Parameters must be root tasks
+            # All other new tasks should be added to the current case (if any)
+            if not isinstance(task, Parameter):
```

Comment:
I put this check here rather than in the `case` class since the check preventing `Parameter` upstream tasks is also handled in the `Flow` class. Made sense to keep things together.